internal/worker/worker.go

```diff
@@ -85,9 +85,13 @@ func (w *BaseWorker) Start(ctx context.Context) error {
 	go func() {
 		for {
 			data, err := w.ioHandler.Read(ctx)
-			if err != nil {
-				readErrChan <- err
-				return
+			if err != nil || len(data) == 0 {
```

Comment:
Could you please add a comment explaining when len(data) can be 0?

Reply:
Another alternative would be ensure that the IO handler returns error instead of data with zero length. wdyt?

Reply:
Sounds good

Reply:
Thanks @soumya-codes for above PR. I will remove this check from this PR